Update V3 status and minor fixes.

MSanKeys963 opened the pull request:
Hi everyone. 👋🏻 

Building this on #260.

The changes in this PR are as follows:

- After going through the recent changes introduced by #260, I realised the date for ZEP0001 acceptance was mentioned as 13th May. The actual date for acceptance is 15th May, when the last vote came in. Check [here](https://github.com/zarr-developers/zarr-specs/issues/227#issuecomment-1547851605) and [here](https://zarr.dev/zeps/accepted/ZEP0001.html).

- Also, we're mentioning [Zarr core development team](https://github.com/orgs/zarr-developers/teams/core-devs) in the specification documents. This shows no team members, and this link throws a 404 to anyone outside our GitHub org. I suggest we remove the hyperlink until we make the teams public, which is visible to everyone (especially users outside the org).

Please have a look and let me know your thoughts.

CC: @zarr-developers/zep1-editors @jhamman

joshmoore commented:
LGTM. Did you look into redirecting from the "draft" URL? (Sorry if that was handled elsewhere?)

joshmoore commented:
Also: we might want to make a list of all of these follow-up steps that are required after a ZEP is accepted (...in ZEP0000?)

MSanKeys963 commented:
> LGTM. Did you look into redirecting from the "draft" URL? (Sorry if that was handled elsewhere?)

Thanks, @joshmoore. I've already added the redirect from [draft](https://zarr.dev/zeps/draft/ZEP0001) → [accepted](https://zarr.dev/zeps/accepted/ZEP0001).

I've just changed the links in the specification document for clarity.

> we might want to make a list of all of these follow-up steps that are required after a ZEP is accepted

Created https://github.com/zarr-developers/zeps/issues/44 for this.

joshmoore reviewed (approved):
LGTM.
